ocean-client/src/vault-maxi.ts

```diff
@@ -76,34 +73,50 @@ export async function main(event: maxiEvent): Promise<Object> {
     }
 
     let vault: LoanVaultActive = vaultcheck
+    let result = true
 
     // 2022-03-08 Krysh: Something went wrong on last execution, we need to clean up, whatever was done
-    if (settings.state !== VaultMaxiState.Start && settings.state !== VaultMaxiState.Finish) {
-        console.log("something went wrong on last execution, we clean up whatever was done")
-        let result = await program.cleanUp(vault, telegram)
-        await store.updateToState(VaultMaxiState.Start)
-        const cleanUpVaultCheck = await program.getVault() as LoanVaultActive
-        await telegram.log("executed clean-up part of script " + (result ? "successfull" : "with problems") + ". vault ratio after clean-up " + cleanUpVaultCheck.collateralRatio)
-        return {
-            statusCode: result ? 200 : 500
+    if (settings.stateInformation && settings.stateInformation.state !== ProgramState.Waiting) {
+        const information = settings.stateInformation
+        console.log("last execution stopped state " + information.state)
+        console.log(" for tx " + information.tx)
+        console.log(" on block height " + information.blockHeight)
+
+        let shouldCleanUp = false
+
+        // 2022-03-09 Krysh: input of kuegi
+        // if we are on state waiting for last transaction and we are still on same block height since last execution
+        // then we should wait for next block
+        const currentBlockHeight = await program.getBlockHeight()
+        if (information.state === ProgramState.WaitingForLastTransaction && information.blockHeight === currentBlockHeight) {
+            console.log("waiting for next block")
+            await program.waitForBlockAfter(currentBlockHeight)
```

Comment:
we wait for block after information.blockHeight, not the current one. cause in block (information.blockHeight + 1) the tx got confirmed.
alternatively we save the txId in the state and directly do a "wait for tx" which would be even better...

Reply:
yeah thought about that too, that we maybe should save txIds. But then I wasn't sure to not make it too big.
And it is called `After` means that the function inside is doing the +1 :D

But I will add storing txIds and checking if this txId is already done and remove that block function